pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -254,60 +258,82 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 		return nil
 	}
 
-	// Get the TaskRun's Pod if it should have one. Otherwise, create the Pod.
-	var pod *corev1.Pod
-	if tr.Status.PodName != "" {
-		pod, err = c.KubeClientSet.CoreV1().Pods(tr.Namespace).Get(tr.Status.PodName, metav1.GetOptions{})
-		if err != nil {
-			c.Logger.Error("Error getting pod %q: %v", tr.Status.PodName, err)
-			return err
-		}
-	} else {
-		pvc, err := c.KubeClientSet.CoreV1().PersistentVolumeClaims(tr.Namespace).Get(tr.Name, metav1.GetOptions{})
-		if errors.IsNotFound(err) {
-			// Create a persistent volume claim to hold Build logs
-			pvc, err = createPVC(c.KubeClientSet, tr)
-			if err != nil {
-				return fmt.Errorf("Failed to create persistent volume claim %s for task %q: %v", tr.Name, err, tr.Name)
-			}
-		} else if err != nil {
-			c.Logger.Errorf("Failed to reconcile taskrun: %q, failed to get pvc %q: %v", tr.Name, tr.Name, err)
-			return err
-		}
-		// Build pod is not present, create build pod.
-		pod, err = c.createBuildPod(ctx, tr, rtr.TaskSpec, rtr.TaskName, pvc.Name)
-		if err != nil {
-			// This Run has failed, so we need to mark it as failed and stop reconciling it
-			var msg string
-			if tr.Spec.TaskRef != nil {
-				msg = fmt.Sprintf("References a Task %s that doesn't exist: ", fmt.Sprintf("%s/%s", tr.Namespace, tr.Spec.TaskRef.Name))
-			} else {
-				msg = fmt.Sprintf("References a TaskSpec with missing information: ")
-			}
-			tr.Status.SetCondition(&duckv1alpha1.Condition{
+	var after *duckv1alpha1.Condition
+	before := tr.Status.GetCondition(duckv1alpha1.ConditionSucceeded)
+
+	if !tr.Status.StartTime.IsZero() && tr.Spec.PipelineRunTimeout != nil {
+		if time.Now().After(tr.Spec.PipelineRunTimeout.Time) {
```

Comment:
could both controller share the timeout logic?

Reply:
It's been yanked out here and is just using `Build`'s timeout.